src/engine/TransitivePath.cpp

```diff
@@ -658,119 +449,158 @@ void TransitivePath::computeTransitivePathRightBound(
+    for (Id otherNode : *hull[node]) {
+      table.emplace_back();
+      table(rowIndex, startSideCol) = node;
+      table(rowIndex, targetSideCol) = otherNode;
 
-  idTable.setNumColumns(getResultWidth());
+      TransitivePath::copyColumns<TEMP_WIDTH, WIDTH>(templateView, table, i,
+                                                     rowIndex, skipCol);
 
-  size_t subWidth = subRes->idTable().numColumns();
-  if (_leftSideTree != nullptr) {
-    shared_ptr<const ResultTable> leftRes = _leftSideTree->getResult();
-    size_t leftWidth = leftRes->idTable().numColumns();
-    CALL_FIXED_SIZE((std::array{subWidth, leftWidth, _resultWidth}),
-                    &TransitivePath::computeTransitivePathLeftBound, this,
-                    &idTable, subRes->idTable(), leftRes->idTable(),
-                    _leftSideCol, _rightIsVar, _leftSubCol, _rightSubCol,
-                    _rightValue, _minDist, _maxDist, _resultWidth);
-  } else if (_rightSideTree != nullptr) {
-    shared_ptr<const ResultTable> rightRes = _rightSideTree->getResult();
-    size_t rightWidth = rightRes->idTable().numColumns();
-    CALL_FIXED_SIZE((std::array{subWidth, rightWidth, _resultWidth}),
-                    &TransitivePath::computeTransitivePathRightBound, this,
-                    &idTable, subRes->idTable(), rightRes->idTable(),
-                    _rightSideCol, _leftIsVar, _leftSubCol, _rightSubCol,
-                    _leftValue, _minDist, _maxDist, _resultWidth);
-  } else {
-    CALL_FIXED_SIZE(subWidth, &TransitivePath::computeTransitivePath, this,
-                    &idTable, subRes->idTable(), _leftIsVar, _rightIsVar,
-                    _leftSubCol, _rightSubCol, _leftValue, _rightValue,
-                    _minDist, _maxDist);
+      rowIndex++;
+    }
   }
-
-  LOG(DEBUG) << "TransitivePath result computation done." << std::endl;
-  // NOTE: The only place, where the input to a transitive path operation is not
-  // an index scan (which has an empty local vocabulary by default) is the
-  // `LocalVocabTest`. But it doesn't harm to propagate the local vocab here
-  // either.
-  return {std::move(idTable), resultSortedOn(), subRes->getSharedLocalVocab()};
 }
 
 // _____________________________________________________________________________
-std::shared_ptr<TransitivePath> TransitivePath::bindLeftSide(
-    std::shared_ptr<QueryExecutionTree> leftop, size_t inputCol) const {
-  return bindLeftOrRightSide(std::move(leftop), inputCol, true);
+template <size_t WIDTH>
+void TransitivePath::fillTableWithHull(IdTableStatic<WIDTH>& table, Map hull,
+                                       size_t startSideCol,
+                                       size_t targetSideCol) {
+  size_t rowIndex = 0;
+  for (auto const& [node, linkedNodes] : hull) {
+    for (Id linkedNode : *linkedNodes) {
+      table.emplace_back();
+      table(rowIndex, startSideCol) = node;
+      table(rowIndex, targetSideCol) = linkedNode;
+
```

Comment:
Can also be optimized in the future as we probably know that the two columsn are either {0, 1} or {1, 0}.